ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,85 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
+isolated function channelWriteCsv(string path, FileWriteOption option, string[][]|map<anydata>[] content) returns Error? {
+
+    WritableCSVChannel csvChannel;
     if content is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
         foreach string[] r in content {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
+        check csvChannel.close();
     } else if content is map<anydata>[] {
+        string[] headers = [];
+        if content.length() == 0 {
+            GenericError e = error GenericError("Input contains an empty array.");
+            return e;
+        }
+        if option == OVERWRITE {
+            headers = content[0].keys();
+            csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
+            Error? headerWriteResult = csvChannel.write(headers);
+            if headerWriteResult is Error {
+                check csvChannel.close();
+                return headerWriteResult;
+            }
+        } else {
+            headers = [];
+            stream<string[], Error?>|Error csvContent = fileReadCsvAsStream(path);
```

Comment:
why are we not handling errors here

Reply:
We didn't handle the errors here as we switched to OVERWRITE mode in the case of any Error, this might not be the best behaviour and this is updated in [b73c9fb](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/b73c9fbca4855aac1c778ff71c05675db4a34b6f). Here we switch to OVERWRITE only in the case of `FileNotFoundError`

Reply:
Why should we return FileNotFoundError in OVERWRITE scenario? AFAIK the io module behaviour is if file is not found we create the file and continue writting

Reply:
Yes, that error is not passed to the user and we handle it in the library. i.e. In the case of APPEND, we have to read the CSV file in order to get the correct order of the elements in the file. If the file is not found, we create a new file and write the content to it.